tests/integration/Core/Content/Product/SalesChannel/CrossSelling/CrossSellingRouteTest.php

```diff
@@ -500,11 +500,40 @@ static function (ProductCrossSellingIdsCriteriaEvent $event) use ($productReposi
         static::assertEquals(5, $element->getCrossSelling()->getAssignedProducts()?->count());
     }
 
-    private function createProductStream(bool $includesIsCloseoutProducts = false, bool $noStock = false): string
+    public function testCrossSellingProductStreamNotContainsProduct(): void
+    {
+        $productId = Uuid::randomHex();
+
+        $productData = $this->getProductData($productId);
+        $productData['crossSellings'] = [[
+            'name' => 'Test Cross Selling',
+            'sortBy' => ProductCrossSellingDefinition::SORT_BY_PRICE,
+            'sortDirection' => FieldSorting::ASCENDING,
+            'active' => true,
+            'limit' => 3,
+            'productStreamId' => $this->createProductStream(false, false, $productId),
+        ]];
+
+        $this->productRepository->create([$productData], $this->salesChannelContext->getContext());
+
+        $result = $this->route->load($productId, new Request(), $this->salesChannelContext, new Criteria())
+            ->getResult();
+
+        $element = $result->first();
+
+        static::assertNotNull($element);
+        static::assertNotContains($productId, $element->getProducts()->getIds());
```

Comment:
Could you please add more detailed assertions here?
example:
We have a product stream with more than one product inside
The result should be to return a collection of product streams without productId!

Reply:
I've added an assertion for the product count.
The assertion with the productId is already there, no?
Also, I've removed the cross selling limit which made the testcase irrelevant.

Reply:
static::assertEquals(5, $element->getProducts()->count()); 
That's what I want. 
Everything looks good to me. However, we have failed the UT. Please check it.

Reply:
please use assertCount :see_no_evil:

Reply:
Yeah, the failing testcase was expecting the product to be loaded into its own cross selling 😄 
I've removed the product from the expected cross selling result.

Reply:
@mitelg Sure, I've changed it for the whole test class.

Reply:
I've added some notNull assertions to fix the phpstan errors as well.